### PR TITLE
[cgal] update to 5.6

### PR DESCRIPTION
**Pull request by lrineau**
This pull-request updates the `cgal` port to [CGAL version 5.6](https://www.cgal.org/2023/07/28/cgal56/).

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by jimwang118** [comment]
You can refer to this [comment](https://github.com/microsoft/vcpkg/issues/30622#:~:text=on%20Apr%204-,You%20can,-see%20the%20running) for logs.

**Comment by lrineau** [comment]
> You can refer to this [comment](https://github.com/microsoft/vcpkg/issues/30622#:~:text=on%20Apr%204-,You%20can,-see%20the%20running) for logs.

Thanks to the discussion. I found out the logs, available [in the artifacts of Azure job](https://dev.azure.com/vcpkg/public/_build/results?buildId=92387&view=artifacts&pathAsName=false&type=publishedArtifacts). It turns out the issue is CGAL-5.6 that is no longer tested on Windows x86 (32 bits). And the error is:
```
D:\installed\x86-windows\include\CGAL/cpp_float.h(30): error C3861: '_BitScanForward64': identifier not found
D:\installed\x86-windows\include\CGAL/cpp_float.h(42): error C3861: '_BitScanReverse64': identifier not found
```

I will try to find out a solution with my fellow CGAL developers...

**Comment by lrineau** [comment]
Done: now the CI checks are fine. I will contribute the patch upstream, now that I have verified that it is correct.

File: ports/cgal/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "cgal",
-  "version": "5.5.3",
+  "version": "5.6",
+  "port-version": 1,
```
Comment (jimwang118) [inline ports/cgal/vcpkg.json:4]:
```suggestion
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Review by jimwang118 — APPROVED** [review]

File: versions/baseline.json [hunk 2]
```diff
@@ -1469,7 +1469,7 @@
       "port-version": 3
     },
     "cgal": {
-      "baseline": "5.5.3",
+      "baseline": "5.6",
       "port-version": 0
```
Comment (jimwang118) [inline versions/baseline.json:1473]:
```suggestion
```
This field does not need to be added when updating a major version. For details, refer to the [version rules](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#versioning).

**Review by jimwang118 — CHANGES_REQUESTED** [review]

File: versions/baseline.json (hunk 2 above)
Comment (lrineau) [reply versions/baseline.json:1473]: That line `"port-version": 0` is added automatically by `./vcpkg x-add-version --all` in `baseline.json`. The port meta-data (in `ports/cgal/vcpkg.json`) no longer have `port-version`.

**Review by jimwang118 — APPROVED** [review]

**Pull request merged**